packages/ui/src/primitives/tailwind/Radio/index.tsx

```diff
@@ -25,82 +25,103 @@ Infinite Reality Engine. All Rights Reserved.
 
 import React from 'react'
 import { twMerge } from 'tailwind-merge'
+import { v4 as uuidv4 } from 'uuid'
 
-export const RadioRoot = ({
-  label,
-  value,
-  onChange,
-  selected,
-  className,
-  labelClassName,
-  disabled,
-  description
-}: {
-  label: string
-  value: string | number
-  onChange: React.ChangeEventHandler<HTMLInputElement>
-  selected: boolean
-  className?: string
-  labelClassName?: string
+import cx from 'clsx'
+
+export interface RadioProps {
   disabled?: boolean
+  defaultChecked?: boolean
```

Comment:
the Radio input is always dependent on a parent component
so we can skip using `defaultChecked`
React recommend either using default value or value
we instead specify `value` to be mandatory

this is creating some situations like the one below: although the value is provided, the input is unchecked

![image](https://github.com/user-attachments/assets/08bf6f21-ce63-48f3-a648-1d186f1317af)

Reply:
Hey @aditya-mitra, thanks for your comments! I think I really don't want to change the API of the component (making value required) out of respect for the refactoring process, to avoid breaking anything. But I saw that the issue is already fixed you need to pass the value to pre-select a option of the radio button